src/common/libioencode/ioencode.c

```diff
@@ -42,7 +43,7 @@ static json_t *data_encode_base64 (const char *stream,
                            "encoding", "base64",
                            "data", dest, n);
     }
-    free (dest);
+    ERRNO_SAFE_WRAP (free, dest);
```

Comment:
`json_pack()` does not set errno on failure so probably need to set it in that case.

If we're guaranteeing that errno is set on failure, then IMHO we should state it in the header file so users know what to expect.

Reply:
> json_pack() does not set errno on failure so probably need to set it in that case.

Ahhh, I was only looking at the `malloc` and `base64_encode()` path in that function.  `json_pack()` could fail too.